consensus/istanbul/core/core.go

```diff
@@ -573,6 +577,7 @@ func (c *core) resetRoundState(view *istanbul.View, validatorSet istanbul.Valida
 		headBlock := c.backend.GetCurrentHeadBlock()
 		newParentCommits = newMessageSet(c.backend.ParentBlockValidators(headBlock))
 	}
+	c.msgSigCache = make(map[string]common.Address)
```

Comment:
QQ: Given that proxies don't mine, are we properly clearing the message cache in proxies?

Reply:
@trianglesphere - IIUC your question correctly, the proxies don't start the Istanbul core engine, and shouldn't be using a message sig cache at all.

Reply:
Looking through the code I think proxies do call `checkValidatorSignature` in https://github.com/celo-org/celo-blockchain/blob/0e08809b755aae0afb41e131a3e5e33816108597/consensus/istanbul/backend/handler.go#L101 I believe that proxies then use the cache.

Reply:
Good point.  The proxies are using a different `checkValidatorSignature` implementation:  https://github.com/celo-org/celo-blockchain/blob/0e08809b755aae0afb41e131a3e5e33816108597/consensus/istanbul/utils.go#L54, so the proxies are never using a cache.

There is some duplicated code as a result, since this PR does copy some of the logic in `istanbul.CheckValidatorSignature` into `core.checkValidatorSignature`.   Seems like the best way to resolve that is to modify `istanbul.CheckValidatorSignature` so that it accepts a signature cache as a parameter, and it will be non-nil when called via istanbul core, but nil when called by the proxy.

Reply:
makes sense. I saw the upper case `CheckValidatorSignature` but didn't realize it was being used in the proxy's `checkValidatorSignature`